src/Sparrow/Json/BlittableJsonDocumentBuilder.cs

```diff
@@ -302,15 +323,167 @@ private bool ReadInternal<TWriteStrategy, TReader>(TReader reader)
                         }
 
                         currentState.State = ContinuationState.ReadPropertyName;
+                        goto case ContinuationState.ReadPropertyName;
+
+                    case ContinuationState.ReadBufferedArrayValue:
+
+                        if (reader.Read() == false)
+                        {
+                            continuationState.Push(currentState);
+                            return false;
+                        }
+
+                        if (state.CurrentTokenType == JsonParserToken.EndArray)
+                        {
+                            if (_state._bufferedSequence?.Count > 0)
+                            {
+                                currentState.State = ContinuationState.CompleteBufferedArray;
+                                goto case ContinuationState.CompleteBufferedArray;
+                            }
+
+                            currentState.State = ContinuationState.CompleteArray;
+                            goto case ContinuationState.CompleteArray;
+                        }
+                        
+                        bool processed = false;
+
+                        // We try to read integers and doubles, if the value is a decimal (which can happen)
+                        // then we bail out because we cannot handle it as a vector (as least for now). 
+                        JsonParserToken current = _state.CurrentTokenType;
+                        switch (current)
+                        {
+                            case JsonParserToken.Integer:
+                                _state.AddBuffered(_state.Long);
+                                processed = true;
+                                break;
+                            case JsonParserToken.Float:
+                                if ((_mode & UsageMode.ValidateDouble) == UsageMode.ValidateDouble)
+                                    _reader.ValidateFloat();
+
+                                var numberString = new ReadOnlySpan<byte>(_state.StringBuffer, _state.StringSize);
+                                if (Utf8Parser.TryParse(numberString, out double value, out int bytesConsumed) == false)
+                                    break;
+
+                                Debug.Assert(bytesConsumed == _state.StringSize);
+
+                                _state.AddBuffered(value);
+                                processed = true;
+                                break;
+                        }
+                        
+                        if (processed)
+                            goto case ContinuationState.ReadBufferedArrayValue; // Successfully read a buffered value, continue buffering
+
+                        // Process the buffered sequence
```

Comment:
```suggestion
                        // Flush the buffered entries, since we cannot write it all as a vector array
                        // we have to write it a standard array, and output all the values we already buffered
```